src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,36 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herb) {
+    if (herb instanceof Herbivore) {
+      herb.health = herb.hidden === false ? herb.health - 50 : herb.health;
+    }
+
+    if (herb.health <= 0) {
```

Comment:
The code for removing the dead herbivores should be inside the Animal class. When the health of any animal reaches 0, it should be removed from the alive array. You can create a separate method for this or include this logic in the setter of the health property. 

Try simplify this whole Carnivore class code to be just:

```
class Carnivore extends Animal {
 bite(target) {
    if (target.hidden === false) {
      target.health -= 50;
    }

    Animal.alive = Animal.alive.filter(animal => animal.health > 0);
  };
}
```